computer-use-demo/computer_use_demo/streamlit.py

```diff
@@ -0,0 +1,364 @@
+"""
+Entrypoint for streamlit, see https://docs.streamlit.io/
+"""
+
+import asyncio
+import base64
+import os
+import subprocess
+from datetime import datetime
+from enum import StrEnum
+from functools import partial
+from pathlib import PosixPath
+from typing import cast
+
+import streamlit as st
+from anthropic import APIResponse
+from anthropic.types import (
+    TextBlock,
+)
+from anthropic.types.beta import BetaMessage, BetaTextBlock, BetaToolUseBlock
+from anthropic.types.tool_use_block import ToolUseBlock
+from streamlit.delta_generator import DeltaGenerator
+
+from computer_use_demo.loop import (
+    PROVIDER_TO_DEFAULT_MODEL_NAME,
+    APIProvider,
+    sampling_loop,
+)
+from computer_use_demo.tools import ToolResult
+
+CONFIG_DIR = PosixPath("~/.anthropic").expanduser()
+API_KEY_FILE = CONFIG_DIR / "api_key"
+STREAMLIT_STYLE = """
+<style>
+    /* Hide chat input while agent loop is running */
+    .stApp[data-teststate=running] .stChatInput textarea,
+    .stApp[data-test-script-state=running] .stChatInput textarea {
+        display: none;
+    }
+     /* Hide the streamlit deploy button */
+    .stDeployButton {
+        visibility: hidden;
+    }
+</style>
+"""
+
+WARNING_TEXT = "⚠️ Security Alert: Never provide access to sensitive accounts or data, as malicious web content can hijack Claude's behavior"
+
+
+class Sender(StrEnum):
+    USER = "user"
+    BOT = "assistant"
+    TOOL = "tool"
+
+
+def setup_state():
```

Comment:
Do all these variables need to be defined in the session state? The state should be reserved for variables which must be preserved across reruns (e.g. `messages`). It's not typically useful for values which are just used within a run (e.g. `hide_images`).